File: AdaptiveTD/AdaptiveTD/AdaptiveTD/EventHandler.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace AdaptiveTD
{
    class EventHandler
    {

        List<Event> events = new List<Event>();

        public List<Event> Events
        {
            get { return events; }
            set { events = value; }
        }

        List<Event> queuedEvents = new List<Event>();


        public void NewRound()
        {
            events.Clear();
            for (int i = 0; i < queuedEvents.Count; i++)
                events.Add(queuedEvents[i]);
            queuedEvents.Clear();
        }

        public void QueueEvent(Event e)
        {
            queuedEvents.Add(e);
        }

        public void Clear()
        {
            queuedEvents.Clear();
            events.Clear();
        }
    }
}

File: LibGDXScene2d/my-gdx-game/src/com/me/mygdxgame/EventHandler.java
package com.me.mygdxgame;

import java.util.ArrayList;
import java.util.List;

public class EventHandler {

	List<Event> events = new ArrayList<Event>();
	List<Event> queuedEvents = new ArrayList<Event>();
	public EventHandler()
	{
		
	}
	
	public void update()
	{
		events.clear();
		for(int i = 0; i<queuedEvents.size(); i++)
			events.add(queuedEvents.get(i));
		queuedEvents.clear();
	}
	
	public void queueEvent(Event e)
	{
		queuedEvents.add(e);
	}
}
